### ISSUE TITLE
Definition of done: Test suites, two independent implementations of each feature?

### ISSUE THREAD

**Issue by jonathanrobie** [bible-technology/scripture-burrito#45]
We should think about what our "definition of done" is for the SB specification.

One criterion that is used in several standardization efforts: at least two independent implementations of each feature.  Any feature that is not implemented at least twice is not needed since interoperability is the whole purpose of a standard. Any feature that is not implemented at least twice has not been adequately tested to show that it is well specified.

Test suites are very helpful for this.

**Comment by mvahowe** [comment]
@jonathanrobie I agree that we should do this before 1.0. This is one reason why I've always argued for starting to write code.

I would suggest setting the bar lower for pre-1.0 releases. I think that, eg, getting SB 0.1 out the door is part of how we eventually end up with multiple implementations by the time we get to 1.0. It's hard to persuade management to make a priority of a standard that doesn't exist yet.

For 0.1, I'd settle for "A set of documentation in something approaching English that covers everything we've thought of, together with at least one example of metadata for each flavor and a RelaxNG schema that validates all those examples."

**Comment by jag3773** [comment]
Great idea @jonathanrobie .  I agree that for now my low bar is that the docs build and the `cibuild.sh` script passes, which at the moment just runs the validation on the examples.

As we are able to craft more tests I'm strongly in favor of adding them to the continuous integration build service.  I think that as we start implementing some of this, even in prototype form, we'll begin to see what exactly needs to be included in the test suites.

**Comment by jonathanrobie** [comment]
0.1 beta is not the SB specification ;->

**Comment by mvahowe** [comment]
I agree. I just wanted to make sure we're not painting ourselves into any corners.

**Comment by jonathanrobie** [comment]
For a final specification, I really would suggest that we strip out features that are not used to exchange data among applications.  Interoperability is the main benefit of a standard.  Complexity and rigidity is are the main costs.  I prefer not to impose these costs where there is no interoperability benefit.
